### DISCUSSION TITLE
Offline installation

### DISCUSSION THREAD

**Discussion by ANISHSAJIKUMAR** [autobase-tech/autobase#186] [General]
Can this script run in offline machine with no internet only acces to Redhat repo . Plz help desperate need to know that

**Comment by vitabaks** [comment]
Yes, such logic is embedded in this playbook since Release 1.1.0 (Oct 31, 2019).

 And to be honest, I was already thinking about removing it, since it complicates the development of the playbook maintenance.

#### Installation method without a proxy server (offline installation).
    By default, it is assumed that you will use a proxy server (`proxy_env` variable) to automatically download the necessary packages and their dependencies from external repositories.
    Сurrently, you can also use your own repository for extra packages. You need to preload all the packages and change URLs.
For your convenience, all URLs have been moved to the vars/[Debian.yml](https://github.com/vitabaks/postgresql_cluster/blob/master/vars/Debian.yml) or vars/[RedHat.yml](https://github.com/vitabaks/postgresql_cluster/blob/master/vars/RedHat.yml) variable files.
 Variables: `etcd_package_repo`, `vip_manager_package_repo`, `confd_package_repo` and `pip_package_repo`, `patroni_pip_requirements_repo`, `patroni_pip_package_repo`.

#### Installation method without a repository (optional).
    If you don't have your own repository .... or you want to create something like a "offline installer package":
You can download the necessary packages into postgresql_cluster/files/ directory. Packages from this directory will be used for installation.
 Variables: `installation_method`, `<variable_name>_file` and `packages_from_file`.

**Reply by vitabaks** [reply]
Attention! If you want to use the installation method "file".
You need to independently determine all the necessary the dependencies of deb packages for your version of the Linux distribution.

**Reply by vitabaks** [reply]
it also turned out that pgbouncer does not yet support the installation method from a file - https://github.com/vitabaks/postgresql_cluster/issues/183

**Reply by vitabaks** [reply]
I will be grateful for your PR
